### PR TITLE
FOGL-7475: Syslog Text search API  integration

### PR DESCRIPTION
**Pull request by hemantgarg923**

### CODE REVIEW

File: src/app/components/core/logs/system-log/system-log.component.html [hunk 1]
```diff
@@ -79,11 +79,11 @@
           <div class="column">
             <label class="label is-small">Search</label>
             <input type="search" class="input is-fullwidth is-small" placeholder="search text on current page"
-              [(ngModel)]="searchTerm" />
+              [(ngModel)]="searchTerm" id="syslog-search" (keyup)="getFilteredSysLog()"/>
```
Comment (praveen-garg) [inline src/app/components/core/logs/system-log/system-log.component.html:82]: Don't we need the css classes?

Keep the placeholder, just remove "on current page"

File: src/app/components/core/logs/system-log/system-log.component.ts [hunk 2]
```diff
@@ -233,4 +236,13 @@ export class SystemLogComponent implements OnInit, OnDestroy {
     this.destroy$.unsubscribe();
     this.subscription.unsubscribe();
   }
+
+  getFilteredSysLog() {
```
Comment (praveen-garg) [inline src/app/components/core/logs/system-log/system-log.component.ts:240]: We are using on keyup with delay of 0.3s? 

Add notif service with name "N 1" and north with "N 123" then If I type "N 1" at normal speed for search

- how many requests are there  a
- also try adding 23 to search txt

File: src/app/services/system-log.service.ts [hunk 3]
```diff
@@ -15,11 +15,12 @@ export class SystemLogService {
   /**
    *  GET | fledge/syslog
    */
-  public getSysLogs(source: String, level: String, limit: Number = 0, offset: Number = 0) {
+  public getSysLogs(source: String, level: String, limit: Number = 0, offset: Number = 0, keyword: string) {
     let params = new HttpParams();
     params = params.set('limit', limit.toString());
     params = params.set('offset', offset.toString());
     params = params.set('nontotals', true);
+    params = params.set('keyword', keyword);
```
Comment (praveen-garg) [inline src/app/services/system-log.service.ts:23]: We should probably add a check if keyword.trim() length  > 0  then params = params.set('keyword', keyword);

File: src/app/components/core/logs/system-log/system-log.component.html [hunk 4]
```diff
@@ -78,12 +78,12 @@
           </div>
           <div class="column">
             <label class="label is-small">Search</label>
-            <input type="search" class="input is-fullwidth is-small" placeholder="search text on current page"
-              [(ngModel)]="searchTerm" />
+            <input type="search" class="input is-fullwidth is-small" placeholder="search text"
+              [(ngModel)]="searchTerm" id="syslog-search" (keyup)="getFilteredSysLog()"/>
```
Comment (praveen-garg) [inline src/app/components/core/logs/system-log/system-log.component.html:82]: change will be better than keyup?

**Review by mshariq-nerd — DISMISSED** [review]

**Review by praveen-garg — DISMISSED** [review]

**Review by mshariq-nerd — APPROVED** [review]

**Review by praveen-garg — APPROVED** [review]

**Review by monika-Dianomic — APPROVED** [review]

**Pull request merged**